diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index a59b3fd..d80ac93 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -985,12 +985,12 @@ var WindowManager = new Lang.Class({
         gesture.connect('activated', Lang.bind(this, this._switchApp));
         global.stage.add_action(gesture);
 
-        let mode = Shell.ActionMode.ALL & ~Shell.ActionMode.LOCK_SCREEN;
-        gesture = new EdgeDragAction.EdgeDragAction(St.Side.BOTTOM, mode);
-        gesture.connect('activated', Lang.bind(this, function() {
-            Main.keyboard.show(Main.layoutManager.bottomIndex);
-        }));
-        global.stage.add_action(gesture);
+        // let mode = Shell.ActionMode.ALL & ~Shell.ActionMode.LOCK_SCREEN;
+        // gesture = new EdgeDragAction.EdgeDragAction(St.Side.BOTTOM, mode);
+        // gesture.connect('activated', Lang.bind(this, function() {
+        //     Main.keyboard.show(Main.layoutManager.bottomIndex);
+        // }));
+        // global.stage.add_action(gesture);
     },
 
     _showPadOsd: function (display, device, settings, imagePath, editionMode, monitorIndex) {
