diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index da36c36..5a478b4 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -305,8 +305,8 @@ var Keyboard = new Lang.Class({
     },
 
     _syncEnabled: function () {
-        this._enableKeyboard = this._a11yApplicationsSettings.get_boolean(SHOW_KEYBOARD) ||
-                               this._lastDeviceIsTouchscreen();
+        this._enableKeyboard = Main.actionMode != Shell.ActionMode.NORMAL;
+
         if (!this._enableKeyboard && !this._keyboard)
             return;
 
