diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index e13c7b724..7995bec8f 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -646,7 +646,7 @@ var Keyboard = new Lang.Class({
     _syncEnabled() {
         let wasEnabled = this._enabled;
         this._enableKeyboard = this._a11yApplicationsSettings.get_boolean(SHOW_KEYBOARD);
-        this._enabled = this._enableKeyboard || this._lastDeviceIsTouchscreen();
+        this._enabled = this._enableKeyboard;
         if (!this._enabled && !this._keyboardController)
             return;
 
@@ -657,8 +657,7 @@ var Keyboard = new Lang.Class({
         else if (!this._enabled)
             this.setCursorLocation(null);
 
-        if (!this._enabled && wasEnabled)
-            Main.layoutManager.hideKeyboard(true);
+        Main.layoutManager.hideKeyboard(true);
     },
 
     _destroyKeyboard() {
diff --git a/js/ui/main.js b/js/ui/main.js
index 2cfe941be..2b1152a26 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -314,7 +314,7 @@ function reloadThemeResource() {
     if (_themeResource)
         _themeResource._unregister();
 
-    _themeResource = Gio.Resource.load(global.datadir + '/gnome-shell-theme.gresource');
+    _themeResource = Gio.Resource.load(global.datadir + '/gnome-shell-theme-xxx.gresource');
     _themeResource._register();
 }
 
