--- /var/abs/extra/mutter/PKGBUILD	2016-11-27 00:40:51.031775576 +0300
+++ PKGBUILD	2016-11-27 01:07:08.977575349 +0300
@@ -3,7 +3,7 @@
 # Maintainer: Ionut Biru <ibiru@archlinux.org>
 # Contributor: Michael Kanis <mkanis_at_gmx_dot_de>
 
-pkgname=mutter
+pkgname=mutter-xxx
 pkgver=3.22.2+1+g5c46094
 pkgrel=1
 pkgdesc="A window manager for GNOME"
@@ -15,10 +15,14 @@
          libxkbcommon-x11 gnome-settings-daemon libgudev libinput)
 makedepends=(intltool gobject-introspection git gnome-common)
 groups=(gnome)
+provides=('mutter')
+conflicts=('mutter')
 options=(!emptydirs)
 _commit=5c46094d67f2e5cdafe6edcfd8c0d5894a2dafc6  # gnome-3-22
-source=("git://git.gnome.org/mutter#commit=$_commit")
-sha256sums=('SKIP')
+source=("git://git.gnome.org/mutter#commit=$_commit"
+       titlebars.patch)
+sha256sums=('SKIP'
+            '6b47dcde16a59332f8612da544adc5ce5fabd5c483468ea0a3bc796d7f117d93')
 
 pkgver() {
   cd $pkgname
@@ -27,6 +31,7 @@
 
 prepare() {
   cd $pkgname
+  patch -p1 -i $srcdir/titlebars.patch
   NOCONFIGURE=1 ./autogen.sh
 }
 
